### ISSUE TITLE
Word marked as completely wrong if extra letter added

### ISSUE THREAD

**Issue by harryparkdotio** [balzss/cli-typer#3]
All words are marked as wrong unless the order matches, rather than if the word is correct. Basically this means that words are checked based on letter index, rather than completeness

<img width="902" alt="screen shot 2018-04-17 at 1 05 45 pm" src="https://user-images.githubusercontent.com/11331772/38846510-113f1362-4240-11e8-8835-c983d0e32225.png">

**Comment by qwhex** [comment]
Hi,
Thanks for the report!

What do you propose as a solution? Jumping to the next word when typing a space and marking the remaining part of the previous word as incorrect? 

Currently we calculate WPM by correct chars / 5, not by actual word count, as the latter could be unfair (more likely to be based on luck).

**Comment by harryparkdotio** [comment]
Ideally words should be matched rather than chars individually, therefore, a .split(" ") on spaces could be performed. Then, if the entire word matches (index by word of words rather than chars), mark as correct.

eg.

```
y = correct (yes)
n = wrong (no)

soil wonder bell rock  six  team oil pay   care
soil wonder taco stone four oil  oil money care
y    y      n    n     n    n    y   n     y
```

**Comment by qwhex** [comment]
Yes, that's the simplest implementation I thought of, my concern is that we lose stats on partially correct words. So I think we should perform the current operation word by word, instead of the whole buffer.

Do you have the time and means to implement it?

**Comment by harryparkdotio** [comment]
Word by word sounds like the correct way to go.

time, not really, means, eh code is a lil hard to follow currently

**Comment by balzss** [comment]
Counting correct words instead of correct chars (with average word length) leads to inconsistent results (maybe you get longer words, etc.) and that's why all major typing practice apps use this approach so thanks for the suggestion but there is no need to implement a word by word counter.
